modules/uncertainty/R/ensemble.R

```diff
@@ -17,36 +17,42 @@
 #--------------------------------------------------------------------------------------------------#
 read.ensemble.output <- function(ensemble.size, pecandir, outdir, start.year, end.year, 
                                  variable, ens.run.ids = NULL) {
+  
+  # load manifest
   if (is.null(ens.run.ids)) {
-    samples.file <- file.path(pecandir, "samples.Rdata")
-    if (file.exists(samples.file)) {
-      samples = new.env()
-      load(samples.file, envir = samples)
-      ens.run.ids <- samples$runs.samples$ensemble
-    } else {
-      stop(samples.file, "not found required by read.ensemble.output")
-    }
+    manifest <- utils::read.csv(file.path(pecandir, "runs_manifest.csv"), stringsAsFactors = FALSE)
+    ens.run.ids <- manifest[manifest$type == "Ensemble", ]
   }
   
   expr <- variable$expression
   variables <- variable$variables
   
   ensemble.output <- list()
-  for (row in rownames(ens.run.ids)) {
-    run.id <- ens.run.ids[row, "id"]
-    PEcAn.logger::logger.info("reading ensemble output from run id: ", format(run.id, scientific = FALSE))
+  
+  for (i in seq_len(nrow(ens.run.ids))) {
+    
+    # Handle column name difference (manifest has 'run_id', legacy 'ens.run.ids' might have 'id')
+    run.id <- if ("run_id" %in% names(ens.run.ids)) ens.run.ids$run_id[i] else ens.run.ids$id[i]
```

Comment:
Would it work to iterate over the IDs directly instead of by index?
```suggestion
  # Handle column name difference (manifest has 'run_id', legacy 'ens.run.ids' might have 'id')
  run_ids <- ens.run.ids$run_id %||% ens.run.ids$id
  for (run_id in run_ids) {
```

Reply:
The index `i` is needed for output list indexing, but we could simplify by extracting run_ids once before the loop; and run_id mean the actual id's which seems confusing;
you suggested `for (run_id in run_ids)` but opted for the index-based approach because `ensemble.output[[as.character(i)]]` uses position based keys not run ids and `seq_along()` provides automatic increment without having to define another variable for manual `idx <- idx + 1`